sycl/plugins/unified_runtime/ur/adapters/level_zero/ur_level_zero_queue.cpp

```diff
@@ -854,8 +854,15 @@ ur_queue_handle_t_::ur_queue_handle_t_(
     bool OwnZeCommandQueue, ur_queue_flags_t Properties, int ForceComputeIndex)
     : Context{Context}, Device{Device}, OwnZeCommandQueue{OwnZeCommandQueue},
       Properties(Properties) {
-  // Set the type of commandlists the queue will use.
-  UsingImmCmdLists = Device->useImmediateCommandLists();
+  // Set the type of commandlists the queue will use when user-selected
+  // submission mode. Otherwise use env var setting and if unset, use default.
+  if (isBatchedSubmission())
+    UsingImmCmdLists = false;
+  else if (isImmediateSubmission())
+    UsingImmCmdLists = true;
+  else
+    UsingImmCmdLists = Device->useImmediateCommandLists();
```

Comment:
the environment variable should have priority over the flags, because that way, the user can force one or the other behavior w/o having to change the code.

Reply:
In this case, the code, if it uses an explicit queue property, should win. Only if the queue property is not used should we rely on env vars.

Reply:
@rdeodhar : but let's say you have the binary for a customer application taht uses batched submission, and you want to see  if a functional or performance bug is gone if you used immediate submission? by having the environment variable always taking priority, you can change the behavior and debug the issue w/o needing a new binary.

same from point of view of user: Imagine the user has a very large codebase, where immediate submission is used, and they want just to see what would happen if batched submission were used. They could go and change all the codebase, but that would mean time refactoring, compiling and executing, while with the environment variable, they can test and see any benefits or problems before investing time changing the code.

Reply:
I appreciate your reasoning but when something is explicitly (hard) coded in the program it should not be subject to over-ride through env vars. Do you know of any other SYCL properties that are subject to over-ride through env vars?
A secondary question is whether the env var has been explicitly set or default. Should even a default setting of the env var over-ride the code? What about explicit setting to what was already the default setting? Should that be considered an explicit setting and over-ride what is in the code?

Reply:
@rdeodhar  for the second question, we would need to have a default value of -1 for the environment variable, so something like this:

```cpp
if (environment variable != -1) {
   set queue mode to value set by environment variable;
} else if (batched mode selected by application) {
   set batched mode
} else if (immediate mode selected by application) {
   set immediate mode
} else {
   set defaults // which currently is for instance immediate mode for PVC and batched for others
}
```

so again, this is mostly about helping with the debug and programmability.  but it is not a hard blocker. If you think it is better to have priority in the setting from application, it is good too.

Reply:
The compiler convention is to honor explicit coding and allow env var control over other behavior. So we'll stick with the way its done now.